FungibleToken.ts

```diff
@@ -115,16 +113,18 @@ export class FungibleToken extends TokenContract {
     canMint.assertTrue()
     this.approve(accountUpdate)
     this.emitEvent("Mint", new MintEvent({ recipient, amount }))
-    this.reducer.dispatch(Int64.fromUnsigned(amount))
+    const circulationUpdate = AccountUpdate.create(this.address, this.deriveTokenId())
+    circulationUpdate.balanceChange = Int64.fromUnsigned(amount)
     return accountUpdate
   }
 
   @method.returns(AccountUpdate)
   async burn(from: PublicKey, amount: UInt64): Promise<AccountUpdate> {
     this.paused.getAndRequireEquals().assertFalse()
     const accountUpdate = this.internal.burn({ address: from, amount })
+    const circulationUpdate = AccountUpdate.create(this.address, this.deriveTokenId())
+    circulationUpdate.balanceChange = Int64.fromUnsigned(amount).mul(Int64.minusOne)
```

Comment:
nit: `Int64.fromUnsigned(amount).negV2()` should be more efficient (in terms of constraints) than using `mul()`

Reply:
Thanks!